### DISCUSSION TITLE
Clock Issues with Ableton

### DISCUSSION THREAD

**Discussion by tarmiricmi** [DISTRHO/Cardinal#767] [Q&A] [answered]
### Version

24.09

### Plugin type

VST3

### OS

Windows 10

### DAW / Host (if applicable)

Ableton 12

### Description

Can't find a way to make Cardinal clocked to the Ableton project. Tried ever clock there is, the light on the clock module usually start blinking in the right tempo, but the BPM number stays the same, and the output as well.
I'm losing my mind, someone please help me out.
I tried connected every output from the host time module but with no results.
Using the p2 mode on the clock kinda works, it halves the BPM, so if Ableton is at 180, the Clocked module shows 90, but it's pretty unstable.

**Comment by dromer** [comment]
Can you share a patch that showcases your struggle?

There are several included patches that show how to integrate with the Host Time module, so maybe check with them how to do this?

[I turned the ticket into a discussion as it is not a Bug but likely a user misunderstanding]

**Reply by tarmiricmi** [reply]
Hey, here's the patch https://we.tl/t-4TiSdi7FO6
Basically I am coming from VCV and I just wanted to set up a default patch.
Thanks for the help.

**Reply by tarmiricmi** [reply]
![image](https://github.com/user-attachments/assets/0e64dc2a-2202-422f-acb2-04761c20492e)

**Reply by dromer** [reply, marked as answer]
In the case of the `Clocked` module you have to set the BPM mode to `P4` and then connect the Step out of `Host Time` module.

Click on the `- MODE +` buttons to switch BPM modes.

**Reply by tarmiricmi** [reply]
Thanks so much, looks like that's it (at least at first glance).

**Comment by topazcoding** [comment]
Hi,  I am struggling with sync issues too, it's happening in cubase unify and ableton.

1st beat is always missing, even when using a midi note to retrigger.

Here is the project to test.
[CardinalSyncTest1.zip](https://github.com/user-attachments/files/18011723/CardinalSyncTest1.zip)

**Reply by dromer** [reply]
I am just guessing that you are using the VST3 version of Cardinal?
(since this is what Cubase supports)

Please be more specific in your setup next time.

**Reply by topazcoding** [reply]
> I am just guessing that you are using the VST3 version of Cardinal? (since this is what Cubase supports)
> 
> Please be more specific in your setup next time.

I tried vst2 and vst3 this issue persists in all the hosts mentioned. 

Live 12/Cubase12/unify

Windows 10.

I also tried just connecting the host tempo module but always the first be is missing.

**Reply by dromer** [reply]
What do you mean "also tried connecting Host Tempo"? This is the only module that will sync to the host.

Very odd that the first beat is missing, please create a bug report instead.

**Reply by topazcoding** [reply]
> What do you mean "also tried connecting Host Tempo"? This is the only module that will sync to the host.
> 
> Very odd that the first beat is missing, please create a bug report instead.

Using host sync direct to Gate-Seq-64 (with a midi note/gate to trigger run). did you try the patch ?

**Comment by topazcoding** [comment]
More testing, the issue is with SEQ-GATE-64, using GoSh Stable16 it's solid. hmmm.

**Comment by topazcoding** [comment]
looks like there was a bug that has now been fixed in impromtu gate-seq-64, not sure if the update made it into Cardinal yet ?

https://community.vcvrack.com/t/impromptu-sequencers-new-bug-or-just-me/19720/12

**Reply by dromer** [reply]
They hard deprecated (removed) SMS16 which will require us to fork so we still have backwards compatibility.

So not yet.

**Reply by topazcoding** [reply]
Ok, thanks for your help on this ;-) best Matt

**Reply by falkTX** [reply]
we can backport a fix if it is simple

according to https://github.com/MarcBoule/ImpromptuModular/issues/67 the issue is actually from Rack and fixed in https://github.com/VCVRack/Rack/commit/70a5413a11437a88e1d5fde2210cc83f8e51386e which we already include.

**Reply by topazcoding** [reply]
The first beat missing is happening with other impromptu sequencers ;-(

**Reply by topazcoding** [reply]
Here is a simple test that works in VCV2(free) but shows the missing 1st beat. 
(hit the run button on clkd) 

Best
Matt ;-) 
[Clock_Working_from_VCV2free.zip](https://github.com/user-attachments/files/18040865/Clock_Working_from_VCV2free.zip)
